File: 2017/day4/part1.cpp
#include <iostream>
#include <set>
#include <sstream>
#include <vector>

std::vector<std::string> tokenize(std::string line)
{
	std::vector<std::string> ret;
	std::stringstream ss(line);
	std::string temp;
	while (std::getline(ss, temp, ' ')) {
		ret.push_back(temp);
	}
	return ret;
}

int main(int argc, char *argv[])
{
	std::string line;
	int total_good = 0;
	while (std::getline(std::cin, line)) {
		auto tokens = tokenize(line);
		std::set<std::string> word_set;
		for (const auto &word : tokens) {
			if (word_set.count(word) == 0) {
				word_set.insert(word);
			}
		}
		if (word_set.size() == tokens.size()) total_good++;
	}
	std::cout << total_good << '\n';
	return 0;
}

File: 2017/day4/part1.swift
import Foundation

func parseLine(_ line: String) -> Bool {
    let words = line.components(separatedBy: " ")
    var wordSet = Set<String>()
    for word in words {
        if wordSet.contains(word) {
            return false
        } else {
            wordSet.insert(word)
        }
    }
    return true
}

func part1(_ input: String) -> Int {
    let inputSplit = input.components(separatedBy: "\n")
    return inputSplit.filter { parseLine($0) }.count
}
